### ISSUE TITLE
Documentation Request for Pagerduty Events API v2 in Alertmanager

### ISSUE THREAD

**Issue by muralikanagala** [prometheus/alertmanager#1854]
**What did you do?**
Trying to configure Alertmanager to use Pagerduty API V2
I am able to send alerts to V1 api using service key but using routing_key is not working.
I have tried adding multiple field to the pagerduty config as defined in the struct https://github.com/prometheus/alertmanager/blob/0788670f81ca3ee750d641f586974660fe0d88db/notify/impl.go#L486 but I am seeing  "field not found in type config.plain" errors.
If i dont specify any i am seeing 400 bad request from Pagerduty.
We need some detailed documentation on how to configure Pagerduty api v2 with Alertmanager.

**What did you expect to see?**
Alerts in Pagerduty

**What did you see instead? Under which circumstances?**
400 bad request in or field not found in type config.plain in Alertmanager logs.

**Environment**
Centos7 

* System information:
Linux 3.10.0-957.5.1.el7.x86_64 x86_64

* Alertmanager version:
alertmanager, version 0.16.2 (branch: HEAD, revision: 308b7620642dc147794e6686a3f94d1b6fc8ef4d)
  build user:       root@1e9a48272b38
  build date:       20190405-12:27:40
  go version:       go1.11.6

* Prometheus version:
prometheus, version 2.8.0 (branch: HEAD, revision: 59369491cfdfe8dcb325723d6d28a837887a07b9)
  build user:       root@4c4d5c29b71f
  build date:       20190312-07:46:58
  go version:       go1.11.5

* Alertmanager configuration file:
```
global:
  resolve_timeout: 5m

route:
  group_by: ['alertname']
  group_wait: 10s
  group_interval: 10s
  repeat_interval: 1h
  receiver: 'test-pager'

receivers:
- name: test-pager
  pagerduty_configs:
  - routing_key: <>
```

* Prometheus configuration file:
```
global:
  scrape_interval:     15s # Set the scrape interval to every 15 seconds. Default is every 1 minute.
  evaluation_interval: 15s # Evaluate rules every 15 seconds. The default is every 1 minute.
  # scrape_timeout is set to the global default (10s).

# Alertmanager configuration
alerting:
  alertmanagers:
  - static_configs:
    - targets:
       - 'localhost:9093'

# Load rules once and periodically evaluate them according to the global 'evaluation_interval'.
rule_files:
  - "alert_rules.yml"
  # - "second_rules.yml"

# A scrape configuration containing exactly one endpoint to scrape:
# Here it's Prometheus itself.
scrape_configs:
  # The job name is added as a label `job=<job_name>` to any timeseries scraped from this config.
  - job_name: 'prometheus'

    # metrics_path defaults to '/metrics'
    # scheme defaults to 'http'.

    static_configs:
    - targets: ['localhost:9090']
```

* Logs:
```
Apr 23 15:03:46 localhost.localdomain alertmanager[2764]: level=error ts=2019-04-23T15:03:46.007910672Z caller=notify.go:332 component=dispatcher msg="Error on notify" err="cancelling notify retry for \"pagerduty\" due to unrecoverable error: unexpected status code 400"
Apr 23 15:03:46 localhost.localdomain alertmanager[2764]: level=error ts=2019-04-23T15:03:46.007954564Z caller=dispatch.go:177 component=dispatcher msg="Notify for alerts failed" num_alerts=1 err="cancelling notify retry for \"pagerduty\" due to unrecoverable error: unexpected status code 400"
Apr 23 15:03:55 localhost.localdomain alertmanager[2764]: level=debug ts=2019-04-23T15:03:55.754287661Z caller=dispatch.go:343 component=dispatcher aggrGroup="{}:{alertname=\"Linux - Server Not Responding or under High Load\"}" msg=flushing alerts="[Linux - Server Not Responding or under High Load[031aca9][active]]" Apr 23 15:03:55 localhost.localdomain alertmanager[2764]: level=debug ts=2019-04-23T15:03:55.756770328Z caller=impl.go:653 msg="Notifying PagerDuty" incident="{}:{alertname=\"Linux - Server Not Responding or under High Load\"}" eventType=trigger
Apr 23 15:03:56 localhost.localdomain alertmanager[2764]: level=debug ts=2019-04-23T15:03:56.028036907Z caller=notify.go:670 component=dispatcher msg="Notify attempt failed" attempt=1 integration=pagerduty receiver=test-pager err="unexpected status code 400"


Apr 23 15:27:40 localhost.localdomain alertmanager[2981]: level=error ts=2019-04-23T15:27:40.424126544Z caller=main.go:337 msg="Loading configuration file failed" file=/opt/prometheus/alertmanager/alertmanager.yml err="yaml: unmarshal errors:\n  line 15: field event_action not found in type config.plain\n  line 16: f
```

**Comment by simonpasquier** [comment]
The PagerDuty receiver configuration is documented at https://prometheus.io/docs/alerting/configuration/#pagerduty_config. Regarding the 400 errors, it might be similar to https://github.com/prometheus/alertmanager/issues/1700 which is fixed by #1701  but it isn't yet available in an official release of AlertManager. Hopefully AlertManager v0.17 should be out sooner than later.

If you feel that AlertManager documentation needs an update, feel free to submit a PR (the sources are [here](https://github.com/prometheus/docs/tree/master/content/docs/alerting)).

**Comment by simonpasquier** [comment]
Closing for now as v0.17 of AlertManager includes #1701 which should ease the troubleshooting.
